conductor-client/src/main/kotlin/com/openlattice/data/ids/PostgresEntityKeyIdService.kt

```diff
@@ -237,162 +261,135 @@ class PostgresEntityKeyIdService(
         return getEntityKeyIds(entityKeys, HashMap(entityKeys.size))
     }
 
-    override fun getEntityKey(entityKeyId: UUID): EntityKey {
-        return loadEntityKeys(setOf(entityKeyId))[entityKeyId]!!
-    }
-
-    override fun getEntityKeys(entityKeyIds: Set<UUID>): MutableMap<UUID, EntityKey> {
-        return loadEntityKeys(entityKeyIds)
+    internal fun loadEntityKeyIds(
+            connection: Connection,
+            entityIds: Map<UUID, Set<String>>,
+            allIdWritten: Boolean = false
+    ): MutableMap<EntityKey, UUID> {
+        val ids = HashMap<EntityKey, UUID>(entityIds.values.sumBy { it.size })
+
+        val ps = connection.prepareStatement(if (allIdWritten) entityKeyIdsSqlAny else entityKeyIdsSqlNotIdWritten)
+        entityIds
+                .forEach { (entitySetId, entityIdValues) ->
+                    ps.setObject(1, entitySetId)
+                    ps.setArray(2, PostgresArrays.createTextArray(connection, entityIdValues))
+                    val rs = ps.executeQuery()
```

Comment:
Oh, I see ... you didn't actually change this function at all. Anyway, still curious about the one query vs many queries.

Reply:
It would be near impossible as you would need to OR them together, which would mean you can't use prepared statements due to one to many relation used in where clause.